components/liquidity/PoolTable.tsx

```diff
@@ -47,39 +47,43 @@ const PoolTable = ({
   const { data: baseAssetUsdPrice } = useAssetUsdPrice(baseAssetId);
   const { data: spotPrices } = useMarketSpotPrices(marketId);
 
-  const tableData: TableData[] = pool?.weights?.map((asset, index) => {
-    let amount: Decimal | undefined;
-    let usdValue: Decimal | undefined;
-    let category: { color?: string; name?: string };
-    const assetId = parseAssetIdString(asset?.assetId);
+  const tableData: TableData[] =
+    pool?.weights?.map((asset, index) => {
+      let amount: Decimal | undefined;
+      let usdValue: Decimal | undefined;
+      let category:
+        | { color?: string | null; name?: string | null }
+        | undefined
+        | null;
+      const assetId = parseAssetIdString(asset?.assetId);
 
-    if (IOBaseAssetId.is(assetId)) {
-      amount = basePoolBalance ?? undefined;
-      usdValue = basePoolBalance?.mul(baseAssetUsdPrice ?? 0);
-      category = { color: "#ffffff", name: metadata?.symbol };
-    } else {
-      amount = new Decimal(balances[index]?.free.toString() ?? 0);
-      usdValue = amount
-        .mul(spotPrices?.get(index) ?? 0)
-        ?.mul(baseAssetUsdPrice ?? 0);
-      category = market?.categories[index];
-    }
+      if (IOBaseAssetId.is(assetId)) {
+        amount = basePoolBalance ?? undefined;
+        usdValue = basePoolBalance?.mul(baseAssetUsdPrice ?? 0);
+        category = { color: "#ffffff", name: metadata?.symbol };
+      } else {
+        amount = new Decimal(balances![index]?.free.toString() ?? 0);
```

Comment:
Probably safer to do this
```suggestion
        amount = new Decimal(balances?.[index]?.free.toString() ?? 0);
```